File: csharp/dsf/dsf/src/pool/ClientPool.cs
namespace com.baidu.dsf.pool
{
    using org.apache.commons.pool2;
    using org.apache.commons.pool2.impl;
    using Thrift;

    /// <summary>
    /// 连接池
    /// @author arksea </summary>
    /// @param <T> thrift client </param>
    public class ClientPool<I> : GenericObjectPool<TServiceClient>, System.IDisposable
    {

        public ClientPool(ClientFactory<I> factory, GenericObjectPoolConfig cfg) : base(factory, cfg)
        {
        }

        public void Dispose()
        {
            if (!this.Closed)
            {
                this.close();
            }
        }

        public virtual string ServiceRegname
        {
            get
            {
                ClientFactory<I> f = (ClientFactory<I>) Factory;
                return f.define.regname;
            }
        }
    }

}

File: java/dsframework/src/main/java/dsf/pool/ClientPool.java
package dsf.pool;

import org.apache.commons.pool2.impl.GenericObjectPool;
import org.apache.commons.pool2.impl.GenericObjectPoolConfig;
import dsf.resource.IDisposable;
import org.apache.commons.pool2.PooledObjectFactory;

/**
 * 连接池
 * @author arksea
 */
public class ClientPool extends GenericObjectPool implements IDisposable {

    public ClientPool(PooledObjectFactory factory, GenericObjectPoolConfig cfg) {
        super(factory, cfg);
    }

    @Override
    public boolean isDisposed() {
        return this.isClosed();
    }

    @Override
    public void dispose() {
        if (!this.isClosed()) {
            this.close();
        }
    }
//    public String getServiceRegname() {
//        ClientFactory f = (ClientFactory) getFactory();
//        return f.define.regname;
//    }
}
